### ISSUE TITLE
Default type for polymorphic slots

### ISSUE THREAD

**Issue by nc163** [ViewComponent/view_component#2174]
## Feature request

I would like to use polymorphic slots in the following way:

```ruby
class ListItemComponent < ViewComponent::Base
  renders_one :visual, types: {
    _: AvatarComponent,  # <- default type
    icon: IconComponent,
  }
end
```

```ruby
<%=
render ListItemComponent.new do |component|
  component.with_visual() 
  # or component.with_visual_icon
end
%>
```

### Motivation

Currently, We have to call it in the following way:

```ruby
<%=
render ListItemComponent.new do |component|
  component.with_visual__
end
%>
```

I would prefer being able to call the default slot type simply as `component.with_visual`

Additionally, it might depend on how components are structured, but I often use inheritance in my components.
In this case, having a default slot would make it simpler to write:
```ruby
class Footer < ViewComponent::Base
class Footer::Admin < Footer

...

renders_one :footer, types: {
  _: Footer,
  admin: Footer::Admin
}

...

component.with_footer  # <- For general pages
component.with_footer_admin  # <- For dashboards
```

**Comment by reeganviljoen** [comment]
I like the idea of a default polymorphic slot, since we do now have default slots aswell, I don't like the idea of using `_` as ten way to denote a default polymorphic slot though, I think it would be better using the `default` keyword

All that said, if anyone wants to pick this up it would be great, unless, @joelhawksley, @camertron, @BlakeWilliams, @boardfish  have any other ideas or reservations about this

**Comment by boardfish** [comment]
Yeah, I suppose `default_type` would help to differentiate from supplying `default` as a potential polymorphic slot type.

I would also consider how this overlaps with `SlotableDefault` — I haven't fully formed my thoughts around what the language should be here but it feels as though `default` as a term could mean a lot of different things in a lot of different contexts. 

We could probably avoid having to think about that by changing the interface to slots, but that's a lot of potential upheaval and I'm not sure how many problems it'd solve relative to those it'd create. It occurred to me just now we could do something like:

```rb
renders_one :visual, IconComponent,
  with_variants: ->(component) do
    component.or_variant(:avatar, AvatarComponent)
    # you could call or_variant with all the same arguments as renders_one
  end
```
